### PR TITLE
Add Ahb feature + fix issue wit AhbLite3ToApb3Bridge + verify overlaping for decoder

### PR DESCRIPTION
**Pull request by Snoopy87**

### CODE REVIEW

File: lib/src/main/scala/spinal/lib/bus/amba3/ahblite/AhbLite3.scala [hunk 1]
```diff
@@ -206,7 +242,7 @@ case class AhbLite3(config: AhbLite3Config) extends Bundle with IMasterSlave {
 
     this.HRDATA     := master.HRDATA
     this.HRESP      := master.HRESP
-    this.HREADY     := master.HREADY
+    this.HREADYOUT  := master.HREADY
```
[the remark this answers is not in the document]
Comment (Snoopy87) [reply lib/src/main/scala/spinal/lib/bus/amba3/ahblite/AhbLite3.scala]: On an AHB interconnect I have several slaves and one of them is a blackbox and has a master interface

File: lib/src/main/scala/spinal/lib/bus/misc/Misc.scala [hunk 2]
```diff
@@ -54,10 +54,30 @@ object SizeMapping{
...
+
+    return hasOverlaps
+
+  }
 }
 
 
 case class SizeMapping(base: BigInt, size: BigInt) extends AddressMapping {
+
+  assert(size > 0, "SizeMapping : size must be greater than 0")
```
[the remark this answers is not in the document]
Comment (Snoopy87) [reply lib/src/main/scala/spinal/lib/bus/misc/Misc.scala]: 👍

File: lib/src/main/scala/spinal/lib/bus/amba3/ahblite/AhbLite3Decoder.scala [hunk 3]
```diff
@@ -150,7 +180,7 @@ class AhbLite3Decoder(ahbLite3Config: AhbLite3Config, decodings: Seq[SizeMapping
   }
 
   val requestIndex = OHToUInt(outputs.map(_.HSEL))
-  val dataIndex    = RegNextWhen(requestIndex, io.input.HREADY)
+  val dataIndex    = RegNextWhen(requestIndex, io.input.HREADYOUT)
```
Comment (Snoopy87) [inline lib/src/main/scala/spinal/lib/bus/amba3/ahblite/AhbLite3Decoder.scala]: not sure about that

**Comment by Dolu1990** [comment]
<3

**Pull request merged**